packages/shared/src/services/assets.ts

```diff
@@ -0,0 +1,17 @@
+import { Knex } from 'knex';
+import { Accountability, TransformationParams, TransformationPreset } from '../types';
+import { AuthorizationService } from './authorization';
+export declare interface AssetsService {
+	knex: Knex;
+	accountability: Accountability | null;
+	authorizationService: AuthorizationService;
+	getAsset(
+		id: string,
+		transformation: TransformationParams | TransformationPreset,
+		range?: any // should be Range from '@directus/drive';
```

Comment:
Trying to implement the services interfaces as suggested in this comment:
> [...] The best we can do is probaly still to create dedicated interfaces for each service inside shared and let the actual service classes implement those. [...]
> _Originally posted by @nickrum in https://github.com/directus/directus/discussions/12563#discussioncomment-3447115_

 One of the issue I ran into was that some services function parameters types come from `@directus/drive` or `@directus/schema` package.

For example, this is the case here with `Range` and `StatResponse`.